OpenRA.Mods.Common/Traits/Power/Player/PowerManager.cs

```diff
@@ -98,6 +102,10 @@ public void UpdateActor(Actor a)
 
 		public void RemoveActor(Actor a)
 		{
+			// Do not remove power from actors that are still in the world
+			if (a.IsInWorld)
```

Comment:
> since removing power of an actor that still is in the world is wrong

One place that calls this method is `Power.OnOwnerChanged`, that's a situation where the actor isn't removed from the world but definitely should be removed from the old owners' PowerManager.

Reply:
But the actor should be removed from the world before any `OnOwnerChanged` callbacks run: https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Game/Actor.cs#L430
Or am I missing something here?

Reply:
Ah, ok. Nevermind then.